### ISSUE TITLE
no-unused-prop-types false positive at UNSAFE_componentWillReceiveProps

### ISSUE THREAD

**Issue by 4muzar** [yannickcr/eslint-plugin-react#2345]
I'm watching weird behavior of eslint: after i renamed `componentWillReceiveProps` to `UNSAFE_componentWillReceiveProps` some props(not all) are highlighted in red.

```
if (creatingSmsPaymentSuccess && 
	creatingSmsPaymentSuccess !== this.props.creatingSmsPaymentSuccess) {
    this.setState({
        someValue: true
    });
}

if (creatingSmsPaymentCompleted && 
	creatingSmsPaymentCompleted !== this.props.creatingSmsPaymentCompleted) {
    this.props.history.replace('some_url')
}
```

In code above `creatingSmsPaymentSuccess` prop is highlighted, but `creatingSmsPaymentCompleted` is OK

eslint: "6.0.1"
eslint-plugin-react": "7.14.2"

**Comment by ljharb** [comment]
What version of react is in your eslint config settings for this plugin?

**Comment by 4muzar** [comment]
@ljharb , thanks! this was the case!
